hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -325,16 +325,13 @@ public static List<HoodieRecord> convertMetadataToFilesPartitionRecords(HoodieCo
                           return map;
                         }
 
-                        int offset = partition.equals(NON_PARTITIONED_NAME)
-                            ? (pathWithPartition.startsWith("/") ? 1 : 0)
-                            : partition.length() + 1;
-                        String filename = pathWithPartition.substring(offset);
+                        String fileName = FSUtils.getFileName(pathWithPartition, partitionStatName);
```

Comment:
Before the change, the `partition` identifier is used, instead of `partitionStatName`.  For a partitioned table, there is no difference; for a non-partitioned table, the `partition` identifier is `.` while `partitionStatName` could be empty or `/`.  The new logic depends on `partitionStatName` instead of `partition` identified, and the file name extracted is not affected.

Reply:
the new util getFileName() uses `partition`, right? why not pass `partition` to check? as how previously done

Reply:
We cannot use `partition` here which is generated by `getPartitionIdentifier(partitionStatName)`, changing the empty relative partition path to `.` partition identifier.  `getFileName()` expects plain relative partition path, instead of the partition identifier used in the metadata table.

Reply:
ok i see. i was confused by the var name `partition`, which should actually be called `partitionIdentifier`